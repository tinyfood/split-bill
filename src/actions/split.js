export const editFriends = friends => ({
  type: 'EDIT_FRIENDS',
  friends
})

export const editTips = tipsPercentage => ({
  type: 'EDIT_TIPS',
  tipsPercentage
})

export const editBill = bill => ({
  type: 'EDIT_BILL',
  bill
})

export const deleteBill = () => ({ type: 'DELETE_BILL' })

export const clearBill = () => ({ type: 'CLEAR_BILL' })

export const calculateTipsAmount = () => ({ type: 'CALCULATE_TIPS_AMOUNT' })

export const calculateTotal = () => ({ type: 'CALCULATE_TOTAL' })
