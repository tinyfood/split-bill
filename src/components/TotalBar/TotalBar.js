import React from 'react'
import { connect } from 'react-redux'
import {
  TotalBarWrapper,
  TotalBarTotal,
  TotalBarFields,
  TotalBarField,
  Text
} from './style'

export const TotalBar = ({
  split: { total, bill, friends, tipsPercentage, tipsAmount }
}) => (
  <TotalBarWrapper>
    <TotalBarTotal>
      <Text medium>Total</Text>
      <Text big>${total}</Text>
    </TotalBarTotal>

    <TotalBarFields>
      <TotalBarField>
        <Text small>Bill</Text>
        <Text small>${bill}</Text>
      </TotalBarField>
      <TotalBarField>
        <Text small>Friends</Text>
        <Text small>{friends}</Text>
      </TotalBarField>
      <TotalBarField>
        <Text small>Tips({tipsPercentage}%)</Text>
        <Text small>${tipsAmount}</Text>
      </TotalBarField>
    </TotalBarFields>
  </TotalBarWrapper>
)

const mapStateToProps = state => {
  return {
    split: state.split
  }
}

export default connect(mapStateToProps)(TotalBar)
