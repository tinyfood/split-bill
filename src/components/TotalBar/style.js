import styled from 'styled-components'

export const TotalBarWrapper = styled.div`
  display: flex;
  margin-bottom: 28px;
  padding: 10px;
  border-radius: 6px;
  background-color: #2ecc71;
  color: #ffffff;
  box-shadow: -4px 32px 24px -16px rgba(46, 204, 112, 0.17);
`
export const TotalBarTotal = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1 1 0;
`

export const TotalBarFields = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1 1 0;
  max-width: 140px;
`

export const TotalBarField = styled.div`
  display: flex;
  justify-content: space-between;
`

export const Text = styled.p`
  margin: 6px;
  font-size: ${props => (props.big ? '28px' : props.medium ? '16px' : '14px')};
  font-weight: ${props => (props.bold ? '700' : '400')};
  text-transform: uppercase;
`
