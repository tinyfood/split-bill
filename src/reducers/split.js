const splitReducerDefaultState = {
  total: undefined,
  bill: undefined,
  friends: undefined,
  tips: undefined
}

export default (state = splitReducerDefaultState, action) => {
  switch (action.type) {
    default:
      return state
  }
}
