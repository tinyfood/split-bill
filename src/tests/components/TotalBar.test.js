import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { TotalBar } from '../../components/TotalBar/TotalBar'
import split from '../fixtures/split'

test('should render TotalBar correctly', () => {
  const wrapper = shallow(<TotalBar split={split} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
