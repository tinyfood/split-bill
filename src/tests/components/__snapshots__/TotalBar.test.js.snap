// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TotalBar correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.p
      medium={true}
    >
      Total
    </styled.p>
    <styled.p
      big={true}
    >
      $
      0
    </styled.p>
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.p
        small={true}
      >
        Bill
      </styled.p>
      <styled.p
        small={true}
      >
        $
        0
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.p
        small={true}
      >
        Friends
      </styled.p>
      <styled.p
        small={true}
      >
        3
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.p
        small={true}
      >
        Tips(
        0
        %)
      </styled.p>
      <styled.p
        small={true}
      >
        $
        0
      </styled.p>
    </styled.div>
  </styled.div>
</styled.div>
`;
